File: frontend/src/types/batch.ts
export interface BatchSubmitResponse {
  job_id: string;
  status: string;
  total_count: number;
}

export interface BatchProgress {
  total: number;
  success: number;
  failed: number;
}

export interface BatchStatusResponse {
  job_id: string;
  status: string;
  progress: BatchProgress;
  eta_seconds?: number;
}

File: backend/app/schemas/batch.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class BatchSubmitResponse:
    job_id: str
    status: str
    total_count: int


@dataclass
class BatchProgress:
    total: int
    success: int
    failed: int


@dataclass
class BatchStatusResponse:
    job_id: str
    status: str
    progress: BatchProgress
    eta_seconds: Optional[int] = None
